runtime/binding-asyncapi/src/main/java/io/aklivity/zilla/runtime/binding/asyncapi/internal/config/AsyncapiHttpProtocol.java

```diff
@@ -264,24 +269,23 @@ private <C> HttpRequestConfigBuilder<C> injectPathParams(
 
     private <C> RouteConfigBuilder<C> injectHttpServerRouteGuarded(
         RouteConfigBuilder<C> route,
-        AsyncapiServerView server)
+        String qname,
+        HttpOptionsConfig options,
+        Asyncapi asyncapi,
+        List<AsyncapiSecurityScheme> securities)
     {
-        if (server.security() != null)
+        if (securities != null && !securities.isEmpty())
         {
-            for (Map<String, List<String>> securityItem : server.security())
+            AsyncapiSecuritySchemeView security =
+                AsyncapiSecuritySchemeView.of(asyncapi.components.securitySchemes, securities.get(0));
+
+            if (isOauthEnabled && "oauth2".equals(security.type()))
             {
-                for (String securityItemLabel : securityItem.keySet())
-                {
-                    if (isJwtEnabled && "jwt".equals(securitySchemes.get(securityItemLabel)))
-                    {
-                        route
-                            .guarded()
-                                .name(guardName)
-                                .inject(guarded -> injectGuardedRoles(guarded, securityItem.get(securityItemLabel)))
-                                .build();
-                        break;
-                    }
-                }
+                route
+                    .guarded()
+                    .name(String.format("%s:%s", qname, options.authorization.name))
```

Comment:
Based on other feedback above, this becomes:
```suggestion
                    .name(options.authorization.qname)
```